crates/fuel-gas-price-algorithm/src/v1/tests/algorithm_v1_tests.rs

```diff
@@ -28,3 +32,60 @@ fn calculate__returns_sum_of_da_and_exec_gas_prices() {
     let expected = starting_exec_gas_price + starting_da_gas_price;
     assert_eq!(expected, actual);
 }
+
+fn _worst_case__correctly_calculates_value(
+    new_exec_price: u64,
+    new_da_gas_price: u64,
+    for_height: u32,
+    block_horizon: u32,
+    exec_price_percentage: u64,
+    da_gas_price_percentage: u64,
+) {
+    // given
+    let algorithm = AlgorithmV1 {
+        new_exec_price,
+        exec_price_percentage,
+        new_da_gas_price,
+        da_gas_price_percentage,
+        for_height,
+    };
+
+    // when
+    let target_height = for_height.saturating_add(block_horizon);
+    let actual = algorithm.worst_case(target_height);
+
+    // then
+    let mut expected_exec_price = new_exec_price;
+    let mut expected_da_gas_price = new_da_gas_price;
+
+    for _ in 0..block_horizon {
+        let change_amount = expected_exec_price
+            .saturating_mul(exec_price_percentage)
+            .saturating_div(100);
+        expected_exec_price = expected_exec_price.saturating_add(change_amount);
+
+        let change_amount = expected_da_gas_price
+            .saturating_mul(da_gas_price_percentage)
+            .saturating_div(100);
+        expected_da_gas_price = expected_da_gas_price.saturating_add(change_amount);
+    }
+
+    let expected = expected_exec_price.saturating_add(expected_da_gas_price);
+
+    dbg!(actual, expected);
+    assert!(actual >= expected);
+}
+
+proptest! {
+    #[test]
+    fn worst_case__correctly_calculates_value(
```

Comment:
We should add another test to check for overflows. See the V0 test and this convo:
https://github.com/FuelLabs/fuel-core/pull/2025#discussion_r1683965151

Reply:
Test added in https://github.com/FuelLabs/fuel-core/pull/2219/commits/6cb31617facd5b55ed99b649146c2750654eb133